strawberry/type.py

```diff
@@ -110,7 +114,20 @@ def _get_fields(wrapped):
             if hasattr(klass, IS_STRAWBERRY_INTERFACE)
         ]
 
+    def copy_with_type(*types):
+        types_replacement_map = dict(
+            zip([param.__name__ for param in wrapped.__parameters__], types)
+        )
+        copied_name = "".join([type.__name__.capitalize() for type in types]) + name
+
+        return TypeClass(
+            copied_name,
+            lambda: _get_fields(wrapped, types_replacement_map=types_replacement_map),
+            **extra_kwargs
+        )
+
     wrapped.field = TypeClass(name, lambda: _get_fields(wrapped), **extra_kwargs)
+    wrapped.copy_with_type = copy_with_type
```

Comment:
we should forbid using the type directly, and only add copy with type when extending a generic